### PR TITLE
[improve](nereids) support pushdown count when scan project without any slot ref

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

```
select 66 from baseall_dup;
could use pushAggOp=COUNT, so no need to scan real data
```
count_ssb

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java [hunk 1]
```diff
@@ -306,6 +316,45 @@ && couldConvertToMulti(agg))
...
+        if (logicalScan != null) {
+            KeysType keysType = logicalScan.getTable().getKeysType();
+            if (keysType != KeysType.DUP_KEYS) {
+                return canNotPush;
+            }
+        }
+        List<Expression> projectExpr = project.getProjects()
+                .stream()
+                .flatMap(p -> Project.collectExpressions(p).stream())
+                .collect(ImmutableList.toImmutableList());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:339]: do not use stream api, it hurt performance

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java [hunk 2]
```diff
@@ -306,6 +316,45 @@ && couldConvertToMulti(agg))
...
+        boolean noSlotRef = projectExpr.stream().allMatch(expr -> {
+            if (expr instanceof SlotReference) {
+                return false;
+            }
+            return true;
+        });
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:345]: use `e.anyMatch(SlotReference.class::isInstance)`

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java [hunk 3]
```diff
@@ -306,6 +317,46 @@ && couldConvertToMulti(agg))
...
+        boolean noSlotRef = true;
+        for (Expression expr : projectExpr) {
+            if (expr instanceof SlotReference) {
+                noSlotRef = false;
+                break;
+            }
+        }
+        if (!noSlotRef) {
+            return canNotPush;
+        }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:350]:
```java
for (Expression e : project.getProjects()) {
    if (e.anyMatch(SlotReference.class::isInstance)) {
        return canNotPush;
    }
}
```
btw, could not push down, if contains Nondeterministic expression, such as UUID()

**Comment by zhangstar333** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**